File: sdks/java/src/main/java/com/apollodeploy/signal/sdk/api/SuppressionsAPI.java
package com.apollodeploy.signal.sdk.api;

import com.fasterxml.jackson.core.type.TypeReference;
import com.apollodeploy.signal.sdk.models.Types.*;
import com.apollodeploy.signal.sdk.transport.Transport;
import java.time.Duration;
import java.util.List;

/** SuppressionsAPI API operations. */
public final class SuppressionsAPI {
    private final Transport transport;

    public SuppressionsAPI(Transport transport) {
        this.transport = transport;
    }

    public SuppressionPageResponse listSuppressions(
        String projectId
    ) {
        String path = "/v1/projects/${projectId}/suppressions";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));

        Transport.Request request = new Transport.Request("GET", path);
        return transport.execute(request, new TypeReference<SuppressionPageResponse>() {
        });
    }

    public void exportSuppressions(
        String projectId
    ) {
        String path = "/v1/projects/${projectId}/suppressions/export";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));

        Transport.Request request = new Transport.Request("GET", path);
        transport.executeVoid(request);
    }

    public SuppressionResponse addSuppression(
        String projectId,
        AddSuppressionBody body
    ) {
        String path = "/v1/projects/${projectId}/suppressions";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));

        Transport.Request request = new Transport.Request("POST", path);
        request.body(body, "application/json");
        return transport.execute(request, new TypeReference<SuppressionResponse>() {
        });
    }

    public void removeSuppression(
        String projectId,
        String email
    ) {
        String path = "/v1/projects/${projectId}/suppressions/${email}";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));
        path = path.replace("${email}", Transport.encodePathSegment(String.valueOf(email)));

        Transport.Request request = new Transport.Request("DELETE", path);
        transport.executeVoid(request);
    }

    public SuppressionImportResponse importSuppressions(
        String projectId,
        SuppressionImportBody body
    ) {
        String path = "/v1/projects/${projectId}/suppressions/import";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));

        Transport.Request request = new Transport.Request("POST", path);
        request.body(body, "application/json");
        return transport.execute(request, new TypeReference<SuppressionImportResponse>() {
        });
    }

}

File: sdks/kotlin/src/main/kotlin/com/apollodeploy/signal/sdk/api/SuppressionsAPI.kt
package com.apollodeploy.signal.sdk.api

import com.apollodeploy.signal.sdk.internal.Transport
import com.apollodeploy.signal.sdk.models.*
import io.ktor.http.HttpMethod
import io.ktor.http.encodeURLPathPart
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.coroutines.flow.Flow

/** SuppressionsAPI API operations. */
class SuppressionsAPI(
    private val transport: Transport,
) {
/** listSuppressions operation. */
    suspend fun listSuppressions(
        projectId: String,
    ): SuppressionPageResponse {
        var path = "/v1/projects/\${projectId}/suppressions"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<SuppressionPageResponse>(
            method = HttpMethod.Get,
            path = path,
            query = query.ifEmpty { null },
            body = null,
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

/** exportSuppressions operation. */
    suspend fun exportSuppressions(
        projectId: String,
    ): Unit {
        var path = "/v1/projects/\${projectId}/suppressions/export"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        transport.executeRaw(
            method = HttpMethod.Get,
            path = path,
            query = query.ifEmpty { null },
            body = null,
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
        return Unit
    }

/** addSuppression operation. */
    suspend fun addSuppression(
        projectId: String,
        body: AddSuppressionBody,
    ): SuppressionResponse {
        var path = "/v1/projects/\${projectId}/suppressions"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<SuppressionResponse>(
            method = HttpMethod.Post,
            path = path,
            query = query.ifEmpty { null },
            body = Json.encodeToJsonElement(body),
            contentType = "application/json",
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

/** removeSuppression operation. */
    suspend fun removeSuppression(
        projectId: String,
        email: String,
    ): Unit {
        var path = "/v1/projects/\${projectId}/suppressions/\${email}"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())
        path = path.replace("\${email}", email.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        transport.executeRaw(
            method = HttpMethod.Delete,
            path = path,
            query = query.ifEmpty { null },
            body = JsonObject(emptyMap()),
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
        return Unit
    }

/** importSuppressions operation. */
    suspend fun importSuppressions(
        projectId: String,
        body: SuppressionImportBody,
    ): SuppressionImportResponse {
        var path = "/v1/projects/\${projectId}/suppressions/import"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<SuppressionImportResponse>(
            method = HttpMethod.Post,
            path = path,
            query = query.ifEmpty { null },
            body = Json.encodeToJsonElement(body),
            contentType = "application/json",
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

}
